Jenkinsfile

```diff
@@ -28,7 +28,7 @@ stage("Unit Test") {
         python -m nltk.downloader all
         make clean
         python setup.py install --force
-        py.test -v --capture=no --durations=0 tests/unittest scripts
+        py.test -v -n auto --durations=0 tests/unittest scripts
```

Comment:
I believe we need to call pytest two times now. Once with `-n auto -m "not serial"` and once with `-n0 -m "serial"`. See https://github.com/pytest-dev/pytest-xdist/issues/84#issuecomment-240608210.

By removing `--capture=no` the output of all passing tests will be removed. If you are confident we never need to look at the output of passing tests I'm happy to remove it. I added it as I wasn't sure if we may run into a situation where the output of previous successful runs would be helpful.

Reply:
OK. Will update the commands.

Currently, the test log is too verbose to navigate. There might be cases where previous logs would be helpful to narrow down a bug. How about we do `--capture=no` for master builds only?

Reply:
It's ok to do `--capture=no` in general. Probably we can just regenerate the output for a particular test at a particular commit in case we need it.

Regarding the update, how about separating the "serial" and "parallel" stages into two Jenkins pipeline stages. In that way the "serial" and "parallel" pytest executions can run in parallel.

Reply:
We are running on only one host right now and `-n auto` would already make use of all CPUs, so we don't need to run these parts in parallel yet.

Reply:
Regarding the output capture, I don't think for each PR the current verbosity is needed, whereas for all master builds it might be useful to provide the output for debugging purpose, hence the suggestion for no capture on master only.